engine/src/main/scala/cromwell/webservice/AkkaHttpService.scala

```diff
@@ -0,0 +1,246 @@
+import akka.http.scaladsl.coding.{Deflate, Gzip, NoCoding}
+import akka.http.scaladsl.server.Route
+import cats.data.NonEmptyList
+import cats.data.Validated.{Invalid, Valid}
+import com.typesafe.config.ConfigFactory
+import cromwell.engine.workflow.WorkflowManagerActor.WorkflowNotFoundException
+import cromwell.engine.workflow.lifecycle.execution.callcaching.CallCacheDiffActor.{BuiltCallCacheDiffResponse, CallCacheDiffActorResponse, FailedCallCacheDiffResponse}
+import cromwell.engine.workflow.lifecycle.execution.callcaching.{CallCacheDiffActor, CallCacheDiffQueryParameter}
+import cromwell.engine.workflow.workflowstore.WorkflowStoreEngineActor.WorkflowStoreEngineAbortResponse
+import lenthall.exception.AggregatedMessageException
+import spray.json.JsObject
+
+import scala.concurrent.duration._
+import scala.util.{Failure, Success, Try}
+
+// FIXME: rename once Ruchi's stuff is in
+trait AkkaHttpService {
+  import cromwell.webservice.AkkaHttpService._
+
+  implicit def actorRefFactory: ActorRefFactory
+  implicit val materializer: ActorMaterializer
+  implicit val ec: ExecutionContext
+
+  val workflowStoreActor: ActorRef
+  val workflowManagerActor: ActorRef
+  val serviceRegistryActor: ActorRef
+
+  // Derive timeouts (implicit and not) from akka http's request timeout since there's no point in being higher than that
+  implicit val duration = ConfigFactory.load().as[FiniteDuration]("akka.http.server.request-timeout")
+  implicit val timeout: Timeout = duration
+
+  val routes =
+    path("workflows" / Segment / "backends") { version =>
+      get { complete(backendResponse) }
+    } ~
+    path("engine" / Segment / "stats") { version =>
+      get {
+        onComplete(workflowManagerActor.ask(WorkflowManagerActor.EngineStatsCommand).mapTo[EngineStatsActor.EngineStats]) {
+          case Success(stats) => complete(stats)
+          case Failure(_) => new RuntimeException("Unable to gather engine stats").failRequest(StatusCodes.InternalServerError)
+        }
+      }
+    } ~
+    path("engine" / Segment / "version") { version =>
+      get { complete(versionResponse) }
+    } ~
+    path("workflows" / Segment / Segment / "status") { (version, possibleWorkflowId) =>
+      get { metadataBuilderRequest(possibleWorkflowId, (w: WorkflowId) => GetStatus(w)) }
+    } ~
+    path("workflows" / Segment / Segment / "outputs") { (version, possibleWorkflowId) =>
+      get { metadataBuilderRequest(possibleWorkflowId, (w: WorkflowId) => WorkflowOutputs(w)) }
+    } ~
+    path("workflows" / Segment / Segment / "logs") { (version, possibleWorkflowId) =>
+      get { metadataBuilderRequest(possibleWorkflowId, (w: WorkflowId) => GetLogs(w)) }
+    } ~
+    path("workflows" / Segment / "query") { version =>
+      (post | get) {
+        parameterSeq { parameters =>
+          extractUri { uri =>
+            metadataQueryRequest(parameters, uri)
+          }
+        }
+      }
+    } ~
+    encodeResponseWith(Gzip, Deflate, NoCoding) {
```

Comment:
I actually had at first - both out of desire (I agree with the belief that unecessarily forcing a content type down someone's throat isn't cool) but also necessity (I couldn't figure out how to replicate the  old behavior for a while). 

This was a relatively late change - I forget what exactly led me to think this but at some point I came to the conclusion that the change would lead to more trouble than it was worth, in the form of people who would like for it to be gzipped and not getting it that way.

I'm more than happy to change it back if people think otherwise, the only "downside" (again if one thinks this way) is that a user would need to explicitly request it to be gzipped instead of by default.